pkg/configuration/configuration.go

```diff
@@ -93,13 +93,20 @@ const (
 	// varDeactivationDomainsExcluded is a string of comma-separated domains that should be excluded from automatic user deactivation
 	// For example: "@redhat.com,@ibm.com"
 	varDeactivationDomainsExcluded = "deactivation.domains.excluded"
+
+	// varForbiddenUsernamePrefixes defines the prefixes that a username may not have when signing up.  If a
+	// username has a forbidden prefix, then the username compliance prefix is added to the username
+	varForbiddenUsernamePrefixes = "username.forbidden.prefixes"
+
+	DefaultForbiddenUsernamePrefixes = "openshift,kubernetes"
 )
 
 // Config encapsulates the Viper configuration registry which stores the
 // configuration data in-memory.
 type Config struct {
-	host         *viper.Viper
-	secretValues map[string]string
+	host                      *viper.Viper
+	secretValues              map[string]string
+	forbiddenUsernamePrefixes []string
```

Comment:
why do we need this var, and why can't we just rely on the config loaded by viper (and still apply the default if the config is explicitly empty, or even better, always ensure that the default values are set)?

Reply:
Ah you're right, I wasn't aware that viper supported array values now.  I've updated to use the viper config directly.